### PR TITLE
`Development`: Remove unused methods from data repositories

### PR DESCRIPTION
**Pull request by pal03377**
#### General

- [X] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [ ] ~Important: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.~
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context

I noticed a lot of `@SuppressWarnings("unused")` annotations on repository interfaces. @krusche confirmed that these annotations shouldn't really be there, so I removed them now. Doing this showed a lot of unused code, which I tracked back to when it was introduced (to avoid any potential mistakes) and then removed.

### Description

I've removed `@SuppressWarnings("unused")` annotations and unused methods. All changes are constrained to be on repository interfaces.

### Steps for Testing

Review the code changes and evaluate whether there might be edge cases where the unused code is really needed. If you are unsure, look at the original PRs for the changes linked below.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### List of removed methods
List of removed methods and why they are now unused:
- `CourseRepository.findWithLecturesAndExamsById` → added in #6073, usage removed in #6318
- `CourseRepository.findStudentGroupName` → added in #2706, usage removed in #6002
- `ExerciseRepository.calculateStatisticsForIndividualCourseExercises` → added in #2869 (initially #2590), usage removed in #6153
- `ExerciseRepository.calculateExerciseStatisticsForIndividualCourseUsingParticipationTable` → added in #2741, usage removed in #6153
- `ExerciseRepository.calculateStatisticsForTeamCourseExercises` → added in #2869 (initially in #2590), usage removed in #6153
- `ExerciseRepository.calculateExerciseStatisticsForTeamCourseExercisesUsingParticipationTable` → added in #2741, usage removed in #6153
- `ExerciseRepository.getExerciseIdsByCourseId` → added in #3077 (initially in #2706), usage removed in #3259
- `ExerciseRepository.findByIdWithCompetenciesBidirectionalElseThrow` → added in #6576, never used
- `LtiOutcomeUrlRepository.findByUserIsCurrentUser` → added in https://github.com/ls1intum/Artemis/commit/0a09a98fb3ab21a42ca1a2a3b6dc3948187c3c08, usage apparently removed in https://github.com/ls1intum/Artemis/commit/d041b99e1dcd9fa71ecb1f1b6a0635a06e122390
- `LtiOutcomeUrlRepository.findByUserIsCurrentUserAndExercise` → added in https://github.com/ls1intum/Artemis/commit/deae29deea7c5274d59dd43412f14e4b33a131cc, usage removed in https://github.com/ls1intum/Artemis/commit/0a09a98fb3ab21a42ca1a2a3b6dc3948187c3c08
- `ModelClusterRepository.findByIdWithEagerElements` → added in #3515, never used
- `ModelClusterRepository.findFeedbacksWithClusterId` → added in #3515, never used
- `ModelElementRepository.findByModelElementIdWithCluster` → added in #3515, never used
- `ModelingSubmissionRepository.findWithEagerResultsFeedbacksAssessorAndParticipationResultsByIdIn` → added in #2725 (initially in #641), removed in #3515
- `ModelingSubmissionRepository.findExampleSubmissionByIdWithEagerResult` → added in #252, usage removed in #4342
- `ModelingSubmissionRepository.countByParticipation_Exercise_Course_IdAndSubmitted` → added in #316 (initially in #126), usage removed in #527
- `MultipleChoiceSubmittedAnswerRepository.findAllWithEagerRelationships/findOneWithEagerRelationships` → difficult to track because of huge commit chains : /
- `ProgrammingExerciseStudentParticipationRepository.findByIdWithAllManualOrSemiAutomaticResultsAndFeedbacksAndRelatedSubmissionAndAssessor` → added in #2831 (initially in #2696), usage removed in #3136
- `ProgrammingExerciseStudentParticipationRepository.findWithIndividualDueDateByExerciseId` → added in #4416, never used (only a method with the same name in `ParticipationRepository`)
- `ResultRepository.findWithEagerSubmissionAndFeedbackAndParticipationByParticipationExerciseId` → added in #4416, usage removed in #4500
- `ResultRepository.findDistinctWithAssessorBySubmissionId` → added in #548, usage removed in #3515
- `ResultRepository.findByIdWithEagerSubmissionAndFeedbacks` → added in #2438, never used
- `ResultRepository.countByAssessor_IdAndParticipation_ExerciseIdAndRatedAndCompletionDateIsNotNull` → added in #453 (initially #316), usage removed in #2866
- `SingleUserNotificationRepository.deleteByRecipientId` → added in #6119, usage removed in #6442
- `StudentParticipationRepository.findByExerciseIdAndStudentLoginAndTestRun` → added in #6262, never used
- `StudentParticipationRepository.findWithEagerSubmissionsByExerciseIdAndStudentLoginAndTestRun` → added in #6262, never used
- `StudentParticipationRepository.findByExerciseIdAndStudentIdAndTestRunWithEagerSubmissionsResultsFeedbacksElseThrow` → added in #5370 (initially #5136), never used
- `StudentParticipationRepository.findWithEagerLegalSubmissionsAndResultsById` → added in #3120 (initially #548), never used since then
- `StudentParticipationRepository.findWithEagerSubmissionsResultsFeedbacksById` → added in #4674, never used since then
- `StudentParticipationRepository.findByStudentIdAndIndividualExercises` → added in #1789, usage removed in https://github.com/ls1intum/Artemis/commit/73b6ba8e56a4917066adf794837eb8222ca4349b
- `TextAssessmentEventRepository.findAllByFeedbackType/findAllByEventType/findAllBySegmentType` → added in #3737, never used
- `TextBlockRepository.findAllByCluster` → added in #660, never used
- `TextBlockRepository.findAllBySubmissionIdIn` → added in #2132, usage removed in #6769
- `TextBlockRepository.countOtherBlocksInClusterBySubmissionId` → added in #3456, usage removed in #6765
- `ConversationParticipantRepository.findConversationParticipantsByUser` → usage added in #6373, never used
- `ConversationParticipantRepository.deleteAllByUser` → added in #2741, usage removed in #6442
- `PostRepository.deleteAllByConversationIn/deleteAllByConversationCreator/deleteAllByAuthor` → added in #6373, usage removed in #6442
- `PostRepository.findPostsByAuthorLogin` → added in #3707, never used

### CODE REVIEW

**Review by jakubriegel — COMMENTED** [review]
What about repositories where `@SuppressWarnings("unused")` was deleted, but no method was removed? Was this annotation accidentally there?

File: src/main/java/de/tum/in/www1/artemis/repository/MultipleChoiceSubmittedAnswerRepository.java [hunk 1]
```diff
@@ -1,25 +1,13 @@
...
 import org.springframework.stereotype.Repository;
 
 import de.tum.in.www1.artemis.domain.quiz.MultipleChoiceSubmittedAnswer;
 
 /**
  * Spring Data JPA repository for the MultipleChoiceSubmittedAnswer entity.
  */
-@SuppressWarnings("unused")
 @Repository
 public interface MultipleChoiceSubmittedAnswerRepository extends JpaRepository<MultipleChoiceSubmittedAnswer, Long> {
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/repository/MultipleChoiceSubmittedAnswerRepository.java:12]: You can delete the whole class, it's empty and unsused.

**Review by Strohgelaender — COMMENTED** [review]
Thanks for keeping the code quality high! I found one place where you can delete the whole class:

**Comment by pal03377** [comment]
> What about repositories where `@SuppressWarnings("unused")` was deleted, but no method was removed? Was this annotation accidentally there?

Apparently, it was accidentally there, yes. I've removed them because they don't do anything besides potentially causing problems in the future.

**Review by jakubriegel — DISMISSED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Review by JohannesStoehr — APPROVED** [review]
Removal looks good and tests still run through

**Pull request merged**